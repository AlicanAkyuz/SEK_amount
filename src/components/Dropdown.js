import React from 'react';
import PropTypes from 'prop-types';

import {
  FullWidthContainer,
  DropdownBox,
  BoldTitle,
  SecondaryText,
} from '../containers/Main/styles';

import sliceNumber from '../utils/sliceNumber';
import dotifyNumber from '../utils/dotifyNumber';
import DropdownList from './DropdownList';

const Dropdown = ({
  notFound, countries, sekAmount, sekRates,
}) => (
  <FullWidthContainer>
    <DropdownBox>
      {notFound ? (
        <BoldTitle>0 Matches</BoldTitle>
      ) : (
        <BoldTitle>
          {/* render different strings acc to found country number */}
          {countries.length > 1 ? `${countries.length} Matches` : `${countries.length} Match!`}
        </BoldTitle>
      )}

      {notFound ? (
        <SecondaryText>No country with that name found</SecondaryText>
      ) : (
        countries.map((country) => {
          // set to SEK value (send as prop to the list) to 'Not found' by default
          let sekValue = `Rates for ${country.currencies[0].code} not found`;

          // change the SEK value if the API has the exchange rate
          if (sekRates[country.currencies[0].code]) {
            // slice the numebr for better UX
            // calculate the total value
            // and add the target country's currency to the string
            sekValue = `${sekAmount} SEK = ${sliceNumber(
              sekRates[country.currencies[0].code] * sekAmount,
            )} ${country.currencies[0].code}`;
          }
          return (
            <DropdownList
              key={country.numericCode}
              name={country.name}
              capital={country.capital}
              population={dotifyNumber(country.population)}
              currencyCode={country.currencies[0].code ? country.currencies[0].code : 'Not Found'}
              sekValue={sekValue} // calculated sekValue or a 'not found' message
            />
          );
        })
      )}
    </DropdownBox>
  </FullWidthContainer>
);

Dropdown.propTypes = {
  notFound: PropTypes.bool.isRequired,
  countries: PropTypes.arrayOf(PropTypes.object).isRequired,
  sekAmount: PropTypes.number.isRequired,
  sekRates: PropTypes.objectOf(PropTypes.number).isRequired,
};

export default Dropdown;
