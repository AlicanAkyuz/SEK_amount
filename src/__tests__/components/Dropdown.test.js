import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Dropdown from '../../components/Dropdown';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Dropdown
      notFound={false}
      countries={[
        {
          name: 'Sweden',
          capital: 'Stockholm',
          population: 100,
          currencies: [{ code: 'SEK' }],
          numericCode: 617298,
        },
      ]}
      sekAmount={130}
      sekRates={{ CAD: 0.97 }}
    />,
    div,
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('correctly renders its required props as defined by prop types', () => {
  const tree = renderer
    .create(
      <Dropdown
        notFound={false}
        countries={[
          {
            name: 'Sweden',
            capital: 'Stockholm',
            population: 100,
            currencies: [{ code: 'SEK' }],
            numericCode: 617298,
          },
        ]}
        sekAmount={130}
        sekRates={{ CAD: 0.97 }}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
