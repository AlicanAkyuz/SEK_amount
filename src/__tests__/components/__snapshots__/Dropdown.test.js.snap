// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly renders its required props as defined by prop types 1`] = `
<div
  className="css-1a821q7"
>
  <ul
    className="css-40fw1n"
  >
    <h1
      className="css-19ozl9x"
    >
      1 Match!
    </h1>
    <div
      className="css-bp3e54"
    >
      <h3
        className="css-qwi5oq"
      >
        Sweden
      </h3>
      <h3
        className="css-f71r11"
      >
        Capital City: Stockholm
      </h3>
      <h3
        className="css-f71r11"
      >
        Population: 100
      </h3>
      <h3
        className="css-f71r11"
      >
        Currency: SEK
      </h3>
      <h3
        className="css-f71r11"
      >
        Rates for SEK not found
      </h3>
    </div>
  </ul>
</div>
`;
